File: Funciones/9. Edad/js/edad.js
/**
 * Calcular la edad de una persona
 * Autor: Santiago GM
 * 03/04/2024
 */
//Como paramétro

let fechaAc;
let fechaNac;
function edades(pfechaAc, pfechaNac) {

   fechaAc = pfechaAc;
   fechaNac = pfechaNac;

  let diferencia;
  let edad;

  diferencia = fechaAc - fechaNac;
  edad = Math.floor(diferencia / (1000 * 60 * 60 * 24 * 365.25));
  return "Tu edad es: " + edad + " años.";
}

//Como Expresión

const edadesExp = function (pfechaAc, pfechaNac) {
   fechaAc = pfechaAc;
   fechaNac = pfechaNac;

  let diferencia;
  let edad;

  diferencia = fechaAc - fechaNac;
  edad = Math.floor(diferencia / (1000 * 60 * 60 * 24 * 365.25));
  return "Tu edad es: " + edad + " años.";
};

File: python/9. Edad/utils/edad.py
# Calcular la edad de una persona
# Autor: Santiago GM

#importar modulos
from datetime import datetime

#Como paramétro

def calcular_edad(fecha_actual_p, fecha_nacimiento_p) :
    edad = fecha_actual_p.year - fecha_nacimiento_p.year
    if (fecha_actual_p.month, fecha_actual_p.day) < (fecha_nacimiento_p.month, fecha_nacimiento_p.day):
        edad -= 1
    return edad


#Como funcion lambda
calcular_edad_lambda = lambda fecha_actual_p, fecha_nacimiento_p: (fecha_actual_p, fecha_nacimiento_p).year - (fecha_nacimiento_p, fecha_actual_p).year if ( fecha_actual_p, fecha_nacimiento_p).month < (fecha_nacimiento_p, fecha_actual_p ).month else (fecha_actual_p, fecha_nacimiento_p).year - (fecha_nacimiento_p , fecha_actual_p).year - 1